packages/core/src/node/console-logger-server.ts

```diff
@@ -59,7 +64,23 @@ export class ConsoleLoggerServer implements ILoggerServer {
     async log(name: string, logLevel: number, message: string, params: any[]): Promise<void> {
         const configuredLogLevel = await this.getLogLevel(name);
         if (logLevel >= configuredLogLevel) {
-            ConsoleLogger.log(name, logLevel, message, params);
+            const fullMessage = ConsoleLogger.log(name, logLevel, message, params);
+            await this.logToFile(fullMessage, params);
+        }
+    }
+
+    protected async logToFile(message: string, params: any[]): Promise<void> {
+        if (this.cli.logFile) {
+            try {
+                const formatted = format(message, ...params) + EOL;
+                await fs.appendFile(this.cli.logFile, formatted);
```

Comment:
It may be overkill, but Theia could create the write stream once with the append flag and push the messages into it instead of always writing the file. I have not tried it, but it may be more efficient.

```js
const s = fs.createWriteStream(this.cli.logFile, { flags: 'a' });
// later
if (!this.failedLogFileWrite) s.push(formatted);
```

Reply:
I agree, using a file stream is probably the better solution. I've adjusted the code to use a file stream.